source/Calamari.Tests/KubernetesFixtures/KubernetesContextScriptWrapperLiveFixtureEks.cs

```diff
@@ -490,17 +540,117 @@ public void DiscoverKubernetesClusterWithInvalidAccountCredentials()
+            variables.Set("Octopus.Action.Aws.Region", region);
+            variables.Set("Octopus.Action.AwsAccount.Variable", account);
+            variables.Set($"{account}.AccessKey", eksClientID);
+            variables.Set($"{account}.SecretKey", eksSecretKey);
+            variables.Set("Octopus.Action.Kubernetes.CertificateAuthority", certificateAuthority);
+            variables.Set($"{certificateAuthority}.CertificatePem", eksClusterCaCertificate);
+        }
+
+        private void SetVariablesForRawYamlCommand()
+        {
+            variables.Set("Octopus.Action.KubernetesContainers.Namespace", "nginx");
+            variables.Set("Octopus.Action.Package.JsonConfigurationVariablesTargets", "**/*.{yml,yaml}");
+
+            variables.AddFeatureToggles(FeatureToggle.MultiGlobPathsForRawYamlFeatureToggle);
+        }
+
+        private void SetVariablesForKubernetesResourceStatusCheck(int timeout)
+        {
+            variables.Set("Octopus.Action.Kubernetes.ResourceStatusCheck", "True");
+            variables.Set("Octopus.Action.KubernetesContainers.DeploymentWait", "NoWait");
+            variables.Set("Octopus.Action.Kubernetes.DeploymentTimeout", timeout.ToString());
+        }
+
+        private static string CreateResourceYamlFile(string directory, string fileName, string content)
+        {
+            var pathToCustomResource = Path.Combine(directory, fileName);
+            File.WriteAllText(pathToCustomResource, content);
+            return pathToCustomResource;
+        }
+
+        private Func<string,string> CreateAddCustomResourceFileFunc(string yamlContent)
+        {
+            return directory =>
+            {
+                CreateResourceYamlFile(directory, ResourceFileName, yamlContent);
+                variables.Set("Octopus.Action.KubernetesContainers.CustomResourceYamlFileName", ResourceFileName);
+                return null;
+            };
+        }
+
+        private Func<string,string> CreateAddPackageFunc(string yamlContent)
+        {
+            const string resourcePackageFileName = "package.1.0.0.zip";
+            return directory =>
+            {
+                var pathToCustomResource =
+                    CreateResourceYamlFile(directory, ResourceFileName, yamlContent);
+                var pathInPackage = Path.Combine("deployments", ResourceFileName);
+                var pathToPackage = Path.Combine(directory, resourcePackageFileName);
+                using (var archive = ZipArchive.Create())
+                {
+                    using (var readStream = File.OpenRead(pathToCustomResource))
+                    {
+                        archive.AddEntry(pathInPackage, readStream);
+                        using (var writeStream = File.OpenWrite(pathToPackage))
+                        {
+                            archive.SaveTo(writeStream, CompressionType.Deflate);
+                        }
+                    }
+                }
```

Comment:
Don't worry about actually changing this, but just FYI:
```suggestion
                using (var archive = ZipArchive.Create())
                using (var readStream = File.OpenRead(pathToCustomResource))
                {
                    archive.AddEntry(pathInPackage, readStream);
                    using var writeStream = File.OpenWrite(pathToPackage); // This is fine as a using declaration because it'll be disposed within this block.
                    archive.SaveTo(writeStream, CompressionType.Deflate);
                }
```